### ISSUE TITLE
How to handle height of item depending of content?

### ISSUE THREAD

**Issue by filoucpp** [BTMorton/angular2-grid#70]
Hi all,

I'm wondering what would be the best solution (if possible) to handle an item that would have a content which height could change (for example a treeview).

Thanks for the hard work, the grid is awesome!
Philippe

**Comment by BTMorton** [comment]
Hmmm... that's a tricky one. It depends on how you're handling the widget config. Whilst the grid checks for updates to the config, the items don't. But what you can do is set a new value to the config. You won't be able to update the properties because that doesn't trigger the setter, but if you did, for example:

```javascript
var tempConfig = this.widgetConfig;
tempConfig.cols = calculateColumns();
this.widgetConfig = null;
setTimeout(() => { this.widgetConfig = tempConfig; });
```

(The setTimeout is to give the angular change detector a chance to update before setting it back).

The other option is:

```javascript
var tempConfig = _.clone(this.widgetConfig);
tempConfig.cols = calculateColumns();
this.widgetConfig = tempConfig;
```

Assuming _.clone is a) a real thing and b) creates a brand new object with the same properties as this.widgetConfig, that isn't an exact clone (i.e. tempConfig !== this.widgetConfig). Unfortunately, that's the only way to do it right now as it's all done using directives, rather than components and I haven't figured out a way to get a reference to the directive to call functions on it.

Does that all make sense?

**Comment by filoucpp** [comment]
Again thanks a lot for the quick answer! It make sense and I will try that tomorrow morning.

If you want to close this feel free as it is not really an issue, I was just asking for advice!
Thanks a lot again :)

**Comment by avernon2** [comment]
Did it work?

On Wednesday, April 27, 2016, Ben Morton <notifications@github.com> wrote:

> Closed #70 <https://github.com/BTMorton/angular2-grid/issues/70>.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/BTMorton/angular2-grid/issues/70#event-643561768>
>


-- 
Sent from my iPhone
